packages/button_driver/include/hardware_test_button/hardware_test_button.py

```diff
@@ -0,0 +1,78 @@
+import rospy
+
+from button_driver import ButtonDriver
+from dt_duckiebot_hardware_tests import HardwareTest, HardwareTestJsonParamType
+
+
+class HardwareTestButton(HardwareTest):
+    def __init__(
+        self,
+        driver: ButtonDriver,
+        led_blink_secs: int = 3,
+        led_blink_hz: int = 1,
+    ) -> None:
+        super().__init__()
+
+        # attr
+        self._driver = driver
+        self._button_released = False
+
+        # test settings
+        self.led_blink_secs = led_blink_secs
+        self.led_blink_hz = led_blink_hz
+
+    def test_id(self) -> str:
+        return "Top button"
+
+    def test_description_preparation(self) -> str:
+        return self.html_util_ul(
+            [
+                "Put your Duckiebot in its normal orientation, and make sure you can see and press the top button."
+            ]
+        )
+
+    def test_description_expectation(self) -> str:
+        return self.html_util_ul(
+            [
+                f"The top button's LED should start blinking at {self.led_blink_hz} HZ.",
+                f"In about {self.led_blink_secs} seconds, it should stop blinking.",
+                "<strong>After</strong> the LED stops blinking, as soon as you press and release the button, the test should finish.",
+                "You should perform the test multiple times and verify the button click event terminates the test promptly.",
+            ]
```

Comment:
You should see the LED in the power button blink for about {self.led_blink_secs} seconds.  When it stops blinking, press and release the button to complete the test.  Try performing this test several times to verify that pressing the power button terminates the test promptly.